### PR TITLE
Feature#23/login functionality

### LINKED ISSUE #23: Login functionality
# REQUIREMENTS
Create the functionality for login view, the user could create a new account or sign in, if there is an existing user, then should create another username or email.

### PR DESCRIPTION
**Pull request by Joshsam31**
## Description 
the aunthenticate for users and sing up works:
- I added a sessions_controller for validate if the user has been login.
- Users_controller has been added for create a new user.
- Sign up view added.
- I made several migrations to change data from a table.

imagen

### CODE REVIEW

File: EasyLearn/app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      render 'new'
+    end
+  end
+   
+  def destroy
+    session[:user_id] = nil
+    flash[:notice] = "You have been logged out."
+    redirect_to login_path
+  end
+end
```
Comment (JuanequeX) [inline EasyLearn/app/controllers/sessions_controller.rb]: Add a blank line here

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: EasyLearn/app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,20 @@
 class ApplicationController < ActionController::Base
+  include ApplicationHelper
+  helper_method :logged_in?
+  helper_method :current_user
+  private 
+  helper_method :current_user, :logged_in?
```
Comment (xFerchoVJ) [inline EasyLearn/app/controllers/application_controller.rb]: You're repeating lines 3 & 4, you can remove them and put this line above the private

File: EasyLearn/app/helpers/registrations_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module RegistrationsHelper
```
Comment (xFerchoVJ) [inline EasyLearn/app/helpers/registrations_helper.rb]: If you don't need this file, you can delete it.

File: EasyLearn/app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                
+                <div class="form-group">
+                  <%= f.label :password, class: "d-flex justify-content-start login_label" %><br/>
+                  <%= f.password_field :password, class: "form-control login_input" %>
+                </div>
+                
+                <div class="form-group">
+                  <%= f.submit "Sign Up", class: "btn btn-dark login_button mb-3" %>
+                </div>
+                <% end %>
```
Comment (xFerchoVJ) [inline EasyLearn/app/views/users/new.html.erb]: Watch out with the tabulation here

**Review by xFerchoVJ — COMMENTED** [review]
I left you some comments, Brody🤓

File: EasyLearn/app/controllers/sessions_controller.rb (hunk 1 above)
Comment (Joshsam31) [reply EasyLearn/app/controllers/sessions_controller.rb]: done.

File: EasyLearn/app/controllers/application_controller.rb (hunk 2 above)
Comment (Joshsam31) [reply EasyLearn/app/controllers/application_controller.rb]: done, love u.

File: EasyLearn/app/helpers/registrations_helper.rb (hunk 3 above)
Comment (Joshsam31) [reply EasyLearn/app/helpers/registrations_helper.rb]: done.

File: EasyLearn/app/views/users/new.html.erb (hunk 4 above)
Comment (Joshsam31) [reply EasyLearn/app/views/users/new.html.erb]: done.

**Review by JuanequeX — APPROVED** [review]
LGTM

File: EasyLearn/app/views/sessions/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+							<hr>
+							<p >Or create a new account <%= link_to "here",signup_path, class: "fw-bold login_a" %></p>
+              
+						</div>
+					</div>
+				</div>
+			</div>
+		</div>
+	</div>
+</section>
```
Comment (AngelBFdev) [inline EasyLearn/app/views/sessions/new.html.erb]: Check the identation tabs, should be 2 spaces

File: EasyLearn/app/views/users/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+              <% end %>
+							<hr>
+							<p >Click <%= link_to "here",login_path, class: "fw-bold login_a" %> to sign in</p>
+						</div>
+					</div>
+				</div>
+			</div>
+		</div>
+	</div>  
+</section>
```
Comment (AngelBFdev) [inline EasyLearn/app/views/users/new.html.erb]: Check the identantion, should be 2 spaces

**Review by AngelBFdev — CHANGES_REQUESTED** [review]

File: EasyLearn/app/views/sessions/new.html.erb (hunk 5 above)
Comment (JuanequeX) [reply EasyLearn/app/views/sessions/new.html.erb]: This identation is correct, but idk why git hub show it of this way

Comment (YilKanda) [reply EasyLearn/app/views/sessions/new.html.erb]: I agree with @Anisbeflo

File: EasyLearn/db/migrate/20220219211841_change_password_to_users.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class ChangePasswordToUsers < ActiveRecord::Migration[7.0]
+  def change
+    remove_column :users, :password
+    #Ex:- change_column("admin_users", "email", :string, :limit =>25)
```
Comment (YilKanda) [inline EasyLearn/db/migrate/20220219211841_change_password_to_users.rb]: Code that you do not use, delete it.

Comment (JuanequeX) [reply EasyLearn/db/migrate/20220219211841_change_password_to_users.rb]: i agree with @YilKanda

File: EasyLearn/app/views/sessions/new.html.erb (hunk 5 above)
Comment (AngelBFdev) [reply EasyLearn/app/views/sessions/new.html.erb]: @JuanequeX Then they need to use this in order to fix it https://www.10bestdesign.com/dirtymarkup/css/

Comment (Joshsam31) [reply EasyLearn/app/views/sessions/new.html.erb]: done

File: EasyLearn/app/views/users/new.html.erb (hunk 6 above)
Comment (Joshsam31) [reply EasyLearn/app/views/users/new.html.erb]: i dont understand, in my local it's different, im sure i use 2 spaces. 
Done.

File: EasyLearn/db/migrate/20220219211841_change_password_to_users.rb (hunk 7 above)
Comment (Joshsam31) [reply EasyLearn/db/migrate/20220219211841_change_password_to_users.rb]: done

**Review by xFerchoVJ — APPROVED** [review]
LGTM

**Review by AngelBFdev — APPROVED** [review]
LGTM

**Pull request merged**